ProcessMaker/processmaker
# Process Templates: Create New Template

## Issue & Reproduction Steps
[FOUR-7376](https://processmaker.atlassian.net/browse/FOUR-7376)

This PR includes several enhancements to enable the creation and management of templates for processes.

## Solution

1. Addition of a new action button labeled 'Create Template' that appears in the Process listings section. This button allows users to easily create a new process template.
2. Includes a 'Create Template' modal  for creating new templates. This modal will be displayed when the 'Create Template' button is clicked.
3. Addition of a new tab labeled 'Templates' within the Process Designer page. This tab enables users to view and manage all existing process templates.
4. Search functionality has been added. Users can now search for process templates.
5. Includes a migration for the `process_templates` table.
6.  Unit tests have been added to validate the functionality of the 'Create Template' feature.


## How to Test

1. Ensure you have Process permissions enabled
2. Navigate to the Process Designer page.
3. Select the 'Create Template'  button to create a new template
4. Once template is created navigate to the 'Templates' tab

**Expected Behavior**
1. Templates are created with no errors.
2. Templates that are created with the same name are suffixed with a  number (x) in the title, when selecting 'Save as New Template'

## Related Tickets & Packages
- Link to any related FOUR tickets, PRDs, or packages

## Code Review Checklist


[FOUR-7376]: https://processmaker.atlassian.net/browse/FOUR-7376?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
